tests/src/Integration/Engine/AnnotationProcessingTest.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Engine\AnnotationProcessingTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Engine;
+
+use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
+use Drupal\Core\Session\SessionManagerInterface;
+
+/**
+ * Tests processing of the ContextDefinition annotation.
+ *
+ * @group rules
+ */
+class AnnotationProcessingTest extends RulesIntegrationTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->enableModule('user');
+    $this->sessionManager = $this->prophesize(SessionManagerInterface::class);
```

Comment:
all class variables should be declared and documented on the class. And why would an annotation test need a session manager? Please add a comment.